core/App/ehsm_core_test.cpp

```diff
@@ -126,144 +126,52 @@ void test_AES128()
-    if (plaintext2.data == NULL) {
-        ret = EH_DEVICE_MEMORY;
-        goto cleanup;
-    }
+    char* returnJsonChar = nullptr;
+    char* plaintext = "TestRSA-3072";
+    char* cmk_base64 = nullptr;
+    char* ciphertext_base64 = nullptr;
+    char* plaintext_base64 = nullptr;
 
-    ret = AsymmetricDecrypt(&cmk, &ciphertext, &plaintext2);
-    if (ret != EH_OK) {
-        printf("Failed(%d) to AsymmetricDecrypt the data\n", ret);
-        goto cleanup;
-    }
-    printf("AsymmetricDecrypt data SUCCESSFULLY!\n");
-    dump_data(plaintext2.data, plaintext2.datalen);
+    printf("============test_RSA3072_encrypt_decrypt start==========\n");
 
-    digest.datalen = 64;
-    digest.data = (uint8_t*)malloc(digest.datalen);
-    if (digest.data == NULL) {
-        ret = EH_DEVICE_MEMORY;
-        goto cleanup;
-    }
-    memset(digest.data, 'B', digest.datalen);
-    dump_data(digest.data, digest.datalen);
-
-    signature.datalen = 0;
-    ret = Sign(&cmk, &digest, &signature);
-    if (ret != EH_OK) {
-        printf("Failed(%d) to get data size of Sign!\n", ret);
+    returnJsonChar = NAPI_CreateKey(EH_RSA_3072, EH_INTERNAL_KEY);
+    if(returnJsonChar == nullptr){
+        printf("NAPI_CreateKey failed!\n");
         goto cleanup;
     }
+    printf("NAPI_CreateKey Json : %s\n", returnJsonChar);
+    printf("Create CMK with RAS SUCCESSFULLY!\n");
 
-    signature.data = (uint8_t*)malloc(signature.datalen);
-    if (signature.data == NULL) {
-        ret = EH_DEVICE_MEMORY;
-        goto cleanup;
-    }
+    cmk_base64 = RetJsonObj::readData_string(returnJsonChar, "cmk_base64");
 
-    ret = Sign(&cmk, &digest, &signature);
-    if (ret != EH_OK) {
-        printf("Failed(%d) to Sign the digest!\n", ret);
+    returnJsonChar = NAPI_AsymmetricEncrypt(cmk_base64, plaintext);
+    if(returnJsonChar == nullptr){
+        printf("NAPI_AsymmetricEncrypt failed!\n");
         goto cleanup;
     }
-    printf("Sign data SUCCESSFULLY!\n");
-    dump_data(signature.data, signature.datalen);
+    printf("NAPI_AsymmetricEncrypt json : %s\n", returnJsonChar);
+    printf("NAPI_AsymmetricEncrypt data SUCCESSFULLY!\n");
 
-    ret = Verify(&cmk, &digest, &signature, &result);
-    if (ret != EH_OK || !result) {
-        printf("Failed(%d) to Verify the signature!\n", ret);
+    ciphertext_base64 = RetJsonObj::readData_string(returnJsonChar, "ciphertext_base64");
+    returnJsonChar = NAPI_AsymmetricDecrypt(cmk_base64, ciphertext_base64);
+    if(returnJsonChar == nullptr){
+        printf("NAPI_AsymmetricDecrypt failed!\n");
         goto cleanup;
     }
-    printf("Verify signature SUCCESSFULLY!\n");
+    printf("NAPI_AsymmetricDecrypt json : %s\n", returnJsonChar);
+    plaintext_base64 = RetJsonObj::readData_string(returnJsonChar, "plaintext_base64");
+    printf("Decrypted plaintext : %s\n",base64_decode(plaintext_base64).c_str());
+    printf("NAPI_AsymmetricDecrypt data SUCCESSFULLY!\n");
```

Comment:
Still need to compare the decrypted text and the plaintext.
Please also update the other tests.